### DISCUSSION TITLE
Able to have fauxmoESP device show up as something other than 'Other'?

### DISCUSSION THREAD

**Discussion by JonS9999** [vintlabs/fauxmoESP#141] [General]
Greetings!

I've been playing around with fauxmoESP for the last fews days, and it's been working great.  Device discovery and registration is working just fine (I'm using various ESPs + a Dot gen 3 + Alexa app on iOS).

One thing I noticed is that it seems that the only way I can add the ESP (via the app) is by going:

   Devices -> (+) -> Add Device -> Other -> Discover Devices

or:

   Devices -> (+) -> Add Device -> Plug -> Other -> Discover Devices

When I look at the info for the device, for Type I see "Other".  I guess this doesn't really surprise me since I had to choose "Other" during the discovery process, but it would be nice if I could change the type to "Plug" or "Light".

<img width="169" alt="2020-12-17 - 002" src="https://user-images.githubusercontent.com/36015625/102537130-2bc86400-4070-11eb-9171-6c8e2df11d91.png">

Is there a way to change the type to something other than Other?

Or should I do the discovery in a different way?  Since I don't own a Hue's hub, I cannot add it by choosing "Philips Hue" as (1) the app asks if the device has Bluetooth (to which I answer 'No'), and (2) the app asks me what type of hub I have.

Thank you for your time, and keep up the good work.  :-)

-Jon

**Comment by pvint** [comment]
There is no way (that I know of) to change the device type in the app.

And, I discovered my devices using an Echo Dot, and it finds them fine, but yes, they show up as "other". I've been looking into this very thing, but unfortunately I haven't been able to dig deep enough out to find if there's a solution.

Note that if you add the device to a group, it will then let you dim it, just in case you're looking for that,

Cheers!
Paul

**Reply by barneyz** [reply]
Paul, you can look at the JSON-string in the github-ESPAlexa-project as an another "Hue-emulation", all devices are recognized as "light devices" (with dimming, also white or color, and on/off/dimming-feedback*) between device and Alexa(-app)

*) switching a Device with it's own tactile switch, i can see the state/dimming-value in the Alexa-app.

**Comment by Dani-Hg** [comment]
Hello since the last update that shouldn't be a problem anymore. Every device is displayed as a lamp and can also be dimmed and switched perfectly via the Alexa app. It also works very well in routines.

Now my question is whether it is possible to add not only lamps but also sockets. For example, I plugged a computer into an electrical outlet. (Of course, it is displayed as a lamp) The problem is when I say: "Alexa switch off lamps", my PC also switches off.

It would be perfect if I could choose between lamp and other device (like before the last update).

Many Thanks

**Comment by JonS9999** [comment]
Hi all.

As a curiosity, has anyone used Wireshark (or similar) to look at the
messages that are sent to/from a real Hue light?  That might clue us in as
to how the JSON needs to be configured to make the ESP look like a real Hue
light.

Just my $0.02 worth...

-Jon

**Comment by pvint** [comment]
@JonS9999 - I'd love to do some captures, but I don't have a real Hue device to test with.

Having said that, as of version 3.2 this should no longer be an issue.
